log_outgoing_requests/utils.py

```diff
@@ -0,0 +1,155 @@
+            except TypeError:
+                return None
+        else:
+            try:
+                content_length = len(http_obj.body)
+            except TypeError:
+                return None
+
+    return content_length
+
+
+def _check_content_length(
+    http_obj: Union[Request, Response],
+    content_length: Optional[int],
+    config: "OutgoingRequestsLogConfig",
+) -> bool:
+    """
+    Check `content_length` against settings.
+
+    If `content_length` could not be determined, the test passes with a
+    warning.
+    """
+
+    if content_length is None:
+        logger.warning(
+            "Content length of the request/response (netloc: %s) could not be determined."
+            % http_obj.netloc
+        )
+        return True
+
+    max_content_length = config.max_content_length
+
+    return content_length <= max_content_length
+
+
+def _check_content_type(
+    content_type: Optional[str], config: "OutgoingRequestsLogConfig"
+) -> bool:
+    """
+    Check `content_type` against settings.
+
+    The content types specified under `LOG_OUTGOING_REQUESTS_CONTENT_TYPES`
+    are split into two groups. For regular types (without wildcard), check
+    if `content_type` is included in the list. For types specified with a
+    wildcard ('*'), check if `content_type` is a substring of any string
+    contained in the list.
+    """
+    if content_type is None:
+        return False
+
+    allowed_mime_types: Iterable[str] = settings.LOG_OUTGOING_REQUESTS_CONTENT_TYPES
+    regulars = [item for item in allowed_mime_types if not item.endswith("*")]
+    wildcards = [item for item in allowed_mime_types if item.endswith("*")]
+
+    if content_type in regulars:
+        return True
+
+    return any(content_type.startswith(pattern[:-1]) for pattern in wildcards)
+
+
+def is_content_admissible_for_saving(
+    http_obj: Union[Request, Response],
+    config: "OutgoingRequestsLogConfig",
+) -> bool:
+    """Determine if content should be saved"""
+
+    content_type = http_obj.headers.get("Content-Type")
+    content_length = _get_content_length(http_obj)
+
+    return _check_content_type(content_type, config) and _check_content_length(
+        http_obj, content_length, config
+    )
+
+
+def get_encoding(request: Request) -> str:
+    """
+    Determine encoding of the `request` (if any).
+
+    Get the `Content-Type` header from the `request` and read the
+    `charset` parameter, if present. Otherwise, fall back on
+    defaults: for text types: ISO-8859-1; for application/json:
+    utf-8; for XML types: utf-8. For all other content types, return
+    the empty encoding.
+
+    This utility is only used for the request; the encoding of the
+    response is determined by the requests library.
+    """
+
+    encoding = ""
+
+    request_content_type = request.headers.get("Content-Type", "")
+
+    if "charset" in request_content_type:
+        encoding = request_content_type.split("charset=")[1]
```

Comment:
I'd approach this like how DRF processes the content-type header for content-type negotation instead of rolling our own: https://github.com/encode/django-rest-framework/blob/master/rest_framework/compat.py#L166